File: 이코테/탐색/이진탐색.py
n, target = map(int, input().split())
arr = list(map(int, input().split()))

## 오름차순 정렬 됐다 가정
def binary_search(arr, target, start, end):
    while start <= end:
        mid = (start + end)//2
        if arr[mid] == target:
            return mid

        if arr[mid] < target:
            start = mid + 1

        if arr[mid] > target:
            end = mid - 1
    return None

result = binary_search(arr, target, 0, len(arr)-1)
if ( result == None ):
    print(" target이 없습니다 ")
else:
    print(arr[result],'(은)는',"[%d] 인덱스 입니다."%result)

File: js/이진탐색.js
function binarySearch(input, target, start, end){
    input = input.sort();
    let mid = parseInt((start + end)/2);    
    if( mid == start || mid == end ) return mid;
    
    if ( input.at(mid) == target ) return mid;

    if ( input.at(mid) > target ) return binarySearch(input, target, start, mid);

    if ( input.at(mid) < target ) return binarySearch(input, target, mid, end);
}

const input = [1,3,5,7,9];
const res = binarySearch(input, 5, 0, input.length-1);
console.log(res);